Content.Client/Chat/ChatManager.cs

```diff
@@ -16,6 +18,20 @@ internal sealed class ChatManager : IChatManager
         private const char OOCAlias = '[';
         private const char MeAlias = '@';
 
+        // Holds any missed messages due to filtering for re-addition to chat later
+<<<<<<< Updated upstream
+        public SortedDictionary<DateTime,MsgChatMessage> filteredHistory = new SortedDictionary<DateTime,MsgChatMessage>();
```

Comment:
You probably should not be storing net messages directly. They have a bunch of extra weight and ideally we would pool them making this use very invalid. Make your own class to store this.

Reply:
I think I'm a bit confused on how to proceed then. Obviously my thinking is that I need to store messages and simply pass them back through `_onChatMessage(MsgChatMessage message)`
whenever I called a `Clear()` on the chat box. I'm thinking a workaround would be to create a more efficient store like a stack or something, but ideally we would want to call to the server for a re-sync down the line? I'm not sure what you mean by pool but I'll do some research.

Reply:
You don't *need* to pass the messages back into `_onChatMessage`. You could just make it call into a different method which chat repopulating would also call back into. You should make your own storage class to pass to that then.

Also don't worry about pooling, that was just a potential reason you shouldn't store `MsgChatMessage` directly.